### PR TITLE
Ports TG's action datums/action button system(mostly for objects)

### PR DESCRIPTION
**Pull request by JustSumBody**
If you ever thought I knew what I was doing prepare to be proven wrong with this PR

Instead of being restricted to one action per item, you can now have multiple by setting them in the item's actions_types list and overriding the object's ui_action_click to do a different thing depending on which button you're clicking. 
You can give them different icons/names/descriptions, and there's probably some neat things you can do now that we also have tooltips. 

While /tg/ uses these for most actions(spells, mechs, and other such things) this is currently just being done with items in mind. 
Also removes the 5 action button limit we apparently had. You can have a lot of action buttons
As it is now, this compiles/runs and works pretty nicely now

Moving them is pretty nifty
actionbutton

Current things that are actually broken:
- [x] The "hide actions" button isn't showing up, can't collapse/reveal the icons
- [x] Ghosting and reentering body/disconnecting and reconnecting causes action buttons to disappear until everything is re-equipped. ~~They also act really weird when they do null the linked action when they reappear from this whoops~~
  ~~Removing the item from its equipped slot and placing it into your hand causes the action button overlay to clear and reappear after a short duration. It's worth noting that it previously disappeared entirely and reappeared, so at least that's consistent. It just looks weirder now since the box stays behind~~
  I'm not sure if this is fixable without changing some more stuff with our HUD code
  Or even that it's that big of a deal, but it really bothers me

Things to do that probably won't break anything:
- [x] Prune remnants of their codebase specific things
- [ ] Replace their icons in the .dmi with relevant stuff (new icons for differentiating jetpack's toggle stabilization button and power on button for instance)
- [x] Convert some more right click object verbs to actions
- [x] find more things to do with this besides objects
- [x] actually figure out what any of this stuff does

### CODE REVIEW

**Comment by ghost** [comment]
Sumguy for collab WHEN

**Comment by clusterfack** [comment]
Pretty fucking good at porting / 10

**Comment by Iamgoofball** [comment]
Oh, don't forget to port the arrow button that allows you to collapse the list so you can hide all the buttons if you don't need them visible.

@ihadtoregisterforthis dude if you guys are enabling support for more than 5 action buttons, make sure you have the hide button available because more than 5 gets kind of screen cluttery

File: code/_onclick/hud/action_button.dm [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	if(hidden)
+		name = "Show Buttons"
+	else
+		name = "Hide Buttons"
+	UpdateIcon()
+	usr.update_action_buttons()
+
+
+/obj/screen/movable/action_button/hide_toggle/proc/InitialiseIcon(mob/living/user)
+	if(isalien(user))
```
Comment (ComicIronic) [inline code/_onclick/hud/action_button.dm:40]: Blech, make this a mob proc.

**Comment by ComicIronic** [comment]
Can I move and toggle the action display?

**Comment by ihadtoregisterforthis** [comment]
Use a better hide button icon, the grey triangle reeks

File: code/_onclick/hud/action_button.dm (hunk 1 above)
Comment (ghost) [reply code/_onclick/hud/action_button.dm:40]: That's really unatomic.

**Comment by JustSumBody** [comment]
@ComicIronic I'm not sure I follow your question, as in move the arrow and toggle it? If so: yes. It hides everything and puts it back in that same location when you unhide it.

@ihadtoregisterforthis I'm artistically challenged. Something similar to how the spellmaster scroll opens/closes rather than the in/out arrow maybe? That could be good.

**Comment by ComicIronic** [comment]
What do you mean by "the same location"? Will action buttons I haven't placed manually show up at the top of the screen again, or next to the arrow?

**Comment by JustSumBody** [comment]
If you haven't moved a button manually it's still at the top of the screen.

**Comment by ComicIronic** [comment]
Haven't you got a var tracking whether or not the action buttons have been placed manually? Seems like you could have the unmoved ones follow the arrow to act more like the spell master behaviour. I'd be surprised as a user to find that I could move the arrow and it did diddly squat to affect where the buttons actually appeared.

**Comment by clusterfack** [comment]
@JustSumBody wheres my action buttons fam

**Comment by clusterfack** [comment]
@JustSumBody pls

**Comment by JustSumBody** [comment]
End of the week FER SURE

**Comment by JustSumBody** [comment]
Didn't mean to push that yet IGNORE THAT

**Comment by JustSumBody** [comment]
Yeah, I can't quite get it to work like that. It's working as "originally intended", but making it function like the spellmasters is a bit beyond me because screen_loc is making my head spin.
I'm not sure if there's another way to do this, whether I should just close this or what.

**Comment by Intigracy** [comment]
confrictu resolved when
PR ready when

**Comment by ComicIronic** [comment]
WHEN

**Comment by ComicIronic** [comment]
Getting a handle on open PR numbers - feel free to reopen when you've made changes.

**Pull request closed without merging**